### ISSUE TITLE
Build not including custom @types

### ISSUE THREAD

**Issue by samuelcastro** [microsoft/TypeScript#29296]
I've created a few custom types in my npm module, which works fine during development mode, however when building up the module with tsc, those custom **@types** are not being released with the build files, and when installing my npm module typescript is complaining about the missing type.

**TS Config**

```ts
{
  "compilerOptions": {
    "outDir": "build/lib",
    "module": "esnext",
    "target": "es5",
    "lib": ["es5", "es6", "es7", "es2017", "esnext", "dom"],
    "sourceMap": true,
    "allowJs": false,
    "moduleResolution": "node",
    "rootDir": "src",
    "baseUrl": "src",
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strict": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
    "declaration": true,
    "esModuleInterop": true,
    "paths": {
      "*": ["../@types/*"]
    }
  },
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "src/setupTests.ts"
  ]
}

```

**Expected behavior:**
Custom @types being included along with node_modules types

**Actual behavior:**
Just node_modules/@types being included

**Comment by DanielRosenwasser** [comment]
Do you have a minimal repro we could clone and observe the behavior in?

**Comment by RyanCavanaugh** [comment]
Closing due to lack of repro
